core/modules/layout/includes/layout.pages.inc

```diff
@@ -45,13 +45,14 @@ function _layout_page_access() {
   $menu_item_name = array_shift($args);
   $menu_item = layout_menu_item_load($menu_item_name);
 
-  // Populate the contexts from the menu system.
-  $contexts = $menu_item->getContexts();
-  foreach ($contexts as $context) {
-    if (isset($context->position) && isset($args[$context->position])) {
-      $context->setData($args[$context->position]);
+  if ($menu_item) {
+    // Populate the contexts from the menu system.
+    $contexts = $menu_item->getContexts();
+    foreach ($contexts as $context) {
+      if (isset($context->position) && isset($args[$context->position])) {
+        $context->setData($args[$context->position]);
+      }
     }
+    return $menu_item->checkAccess();
   }
-
-  return $menu_item->checkAccess();
```

Comment:
Could we make this function return FALSE if no menu item was found? That will return a consistent Boolean value instead of NULL sometimes.